libp2p/muxers/mplex/mplex.nim

```diff
@@ -191,16 +198,16 @@ method close*(m: Mplex) {.async, gcsafe.} =
     return
 
   trace "closing mplex muxer", oid = m.oid
+  await all(
+    toSeq(m.remote.values).mapIt(it.reset()) &
+      toSeq(m.local.values).mapIt(it.reset()))
 
-  checkFutures(
-    await allFinished(
-      toSeq(m.remote.values).mapIt(it.reset()) &
-        toSeq(m.local.values).mapIt(it.reset())))
-
-  checkFutures(await allFinished(m.handlerFuts))
+  await all(m.conns.mapIt(it.close())) # dispose of channe's connections
```

Comment:
so if close raises, we should not wait for the handlers? what should the caller do in response to the exception, now that mplex is partially closed?

Reply:
we've already reset the channel above, conns should't raise one because `conn.close` doesn't race unless it's a Defect, two because close here will simply dispose of the wrapper connection, thats it. That said, it's not a bad idea to wrap mplex'es close in a try/catch/finally as well.

Reply:
these sort of assumptions are best described in a comment, since the compiler doesn't provide any guarantees / help - one can imagine a construct that turns any exception coming out of anything that shouldn't raise into a `Defect`